### PR TITLE
chore: replace lodash with native utilities

### PR DESCRIPTION
**Pull request by eglitise**
This also results in an almost 10x reduction in package size (https://pkg-size.dev/teen_process).

### CODE REVIEW

File: lib/exec.ts [hunk 1]
```diff
@@ -64,7 +63,7 @@ export async function exec<T extends TeenProcessExecOptions = TeenProcessExecOpt
     maxStderrBufferSize: MAX_BUFFER_SIZE,
   };
 
-  const opts = _.defaults({}, originalOpts, defaults) as T;
+  const opts = Object.assign({}, defaults, originalOpts) as T;
```
Comment (mykola-mokhnach) [inline lib/exec.ts]: I think this might be a breaking change, since `defaults` only replaces unset values, while Object.assign replaces all of them unconditionally

File: test/exec-specs.ts [hunk 2]
```diff
@@ -126,10 +125,10 @@ describe('exec', function () {
     const cmd = getFixture('echo.sh');
     const echo1 = 'my name is bob';
     const {stdout, stderr, code} = await exec(cmd, [echo1], {isBuffer: true});
-    expect(_.isString(stdout)).to.be.false;
-    expect(_.isBuffer(stdout)).to.be.true;
-    expect(_.isString(stderr)).to.be.false;
-    expect(_.isBuffer(stderr)).to.be.true;
+    expect(typeof stdout).to.not.equal('string');
```
Comment (mykola-mokhnach) [inline test/exec-specs.ts]: I believe chai has better assertions to verify variable type, e.g. `expect(stdout).to.not.be.a('string');`

Comment (mykola-mokhnach) [reply test/exec-specs.ts]: same below

File: lib/exec.ts (hunk 1 above)
Comment (eglitise) [reply lib/exec.ts]: I'm not so sure. `_.defaults` requires the set of default values to be the last argument, so it makes sense for it to then only replace unset values. `Object.assign` has the default values as the first (or in this case second) argument, with the custom values only being assigned afterwards, so the end result is that only the values not defined in `originalOpts` will be set to a value in `defaults`.

Comment (mykola-mokhnach) [reply lib/exec.ts]: I was just thinking about cases like `defaults = {x: 1}` and `originalOpts = {x: undefined}` 

Object.assign will result in {x: undefined} while defaults in {x: 1}.

Maybe it's an edge case, but this is still a general purpose lib, so we don't know all possible usage scenarious

File: test/exec-specs.ts [hunk 3]
```diff
@@ -139,8 +138,8 @@ describe('exec', function () {
 
     it('should allow binary output', async function () {
       const {stdout} = await exec('cat', [getFixture('screenshot.png')], {encoding: 'binary'});
-      expect(_.isString(stdout)).to.be.true;
-      expect(_.isBuffer(stdout)).to.be.false;
+      expect(stdout).to.be.a('string');
+      expect(Buffer.isBuffer(stdout)).to.be.false;
```
Comment (mykola-mokhnach) [inline test/exec-specs.ts]: this could also be changed to `expect(stdout).to.not.be.instanceOf(Buffer);`

File: test/subproc-specs.ts [hunk 4]
```diff
@@ -164,7 +163,7 @@ describe('SubProcess', function () {
       });
       await subproc.start();
       await new Promise((resolve) => setTimeout(resolve, 50));
-      expect(_.isString(output[0])).to.be.true;
+      expect(typeof output[0]).to.equal('string');
```
Comment (mykola-mokhnach) [inline test/subproc-specs.ts]: same as above

File: test/exec-specs.ts (hunk 3 above)
Comment (mykola-mokhnach) [reply test/exec-specs.ts]: another option: `expect(stdout).to.not.satisfy(Buffer.isBuffer);`

File: lib/subprocess.ts [hunk 5]
```diff
@@ -125,20 +124,26 @@ export class SubProcess<
...
       if (!this.opts.detached) {
         throw new Error(`Unable to detach process that is not started with 'detached' option`);
       }
       detach = true;
       detector = genericStartDetector;
-    } else if (_.isBoolean(timeoutMs) && timeoutMs) {
+    } else if (
+      (typeof timeoutMs === 'boolean' || (timeoutMs as any) instanceof Boolean) &&
+      Boolean(timeoutMs)
+    ) {
```
[the remark this answers is not in the document]
Comment (eglitise) [reply lib/subprocess.ts:146]: This implementation matches the previous lodash-based behavior

File: lib/exec.ts [hunk 6]
```diff
@@ -64,9 +63,14 @@ export async function exec<T extends TeenProcessExecOptions = TeenProcessExecOpt
     maxStderrBufferSize: MAX_BUFFER_SIZE,
   };
 
-  const opts = _.defaults({}, originalOpts, defaults) as T;
+  const normalizedOriginalOpts =
```
Comment (mykola-mokhnach) [inline lib/exec.ts]: this logic could be extracted to a helper function for better readability

File: lib/exec.ts [hunk 7]
```diff
@@ -106,7 +112,7 @@ export async function exec<T extends TeenProcessExecOptions = TeenProcessExecOpt
 
       stream.on('data', (chunk: Buffer) => {
         buffer.add(chunk);
-        if (opts.logger?.debug && _.isFunction(opts.logger.debug)) {
+        if (opts.logger?.debug && typeof opts.logger.debug === 'function') {
```
Comment (mykola-mokhnach) [inline lib/exec.ts]: could be simplified to `typeof opts.logger?.debug === 'function'`

**Review by mykola-mokhnach — APPROVED** [review]
